File: 2015/dia01/main.py
input = open("input.txt", "r").read().strip()
# print(input)


def parte_1():
    map = {"(": 1, ")": -1}
    floor = 0
    for char in input:
        floor += map[char]
    print(floor)


def parte_2():
    map = {"(": 1, ")": -1}
    floor = 0
    position = 0
    for char in input:
        floor += map[char]
        position += 1
        if floor == -1:
            break
    print(position)


if __name__ == "__main__":
    parte_1()
    parte_2()

File: 2015/dia01/cpp/main.cpp
#include <fstream>
#include <print>
#include <vector>

void parte_1(std::string line) {
    int floor{0};
    for (char c : line) {
        if (c == '(') ++floor;
        if (c == ')') --floor;
    }
    std::print("{}\n", floor);
};

void parte_2(std::string line) {
    int floor{0};
    int position{0};
    for (char c : line) {
        if (c == '(') ++floor;
        if (c == ')') --floor;
        ++position;
        if (floor == -1) break;
    }
    std::print("{}\n", position);
};

int main(int argc, char** argv) {
    std::print("\n");

    std::ifstream input;
    input.open(argv[1]);

    std::string line;
    std::vector<std::string> lines;
    while (std::getline(input, line)) {
        lines.push_back(line);
    }
    input.close();

    parte_1(lines[0]);
    parte_2(lines[0]);

    return 0;
}
